### ISSUE TITLE
[NetworkGraph] Amibiguity on graph scope

### ISSUE THREAD

**Issue by nemesisdesign** [interop-dev/netjson#49]
The RFC is not clear on whether NetworkGraph should contain the full network or just what is known to a node.

@fhuberts was deceived by this ambiguity with the first version of the olsrd1 netjson plugin.

I think it would be acceptable to mandate that every implementation returns all the information it has about the network graph. This should be ok for distance vector too.

Another proposal related to this one is #25 (how to distinguish between full graph and partial/local graph?).

CC: @hrogge @gabri94

**Comment by fhuberts** [comment]
>Another proposal related to this one is #25 (how to distinguish between full graph and partial/local graph?).

Add a boolean to the NetworkGraph object, like 'global' or 'local'

**Comment by HRogge** [comment]
I think this is not about "what a router knows"... because olsrd definitely knows about links of remote nodes (through the TC messages).

it is more about direct vs. indirect knowledge.

Yes, a boolean "local" which can be true/false would improve the data returned via NetJSON.

**Comment by gabri94** [comment]
I agree, with a boolean we can determine what the json is containing, and we avoid inconsistency.

In the case of the flag set to boolean, what's the meaning of the list of all the nodes?
I propose to return just the 1-hop neighbor and the 1-hop links in the case of "local"
In the case of "global" we return the list of all the nodes of the network and all the links.

**Comment by HRogge** [comment]
There is no real "input parameter" for NetJSON... so I would say a protocol should ALWAYS put all information it has into the NetJSON object... and tag it with the "local true/false" tag so that the processing knows what to do with it.

**Comment by fhuberts** [comment]
The 'local' flag would mean "this information does not contain the whole network"

So the client know that is must query all nodes

**Comment by fhuberts** [comment]
So we could also name the flag something like 'containWholeNetwork' :-)

**Comment by HRogge** [comment]
This might be something to think about for DistanceVector protocols... Linkstate protocols always have the whole graph.

**Comment by nemesisdesign** [comment]
I also would prefer `NetworkGraph` to be intended as a full network graph unless a `local` boolean attribute indicates otherwise.

**Comment by nemesisdesign** [comment]
After discussing, me, @HRogge and @fhuberts think `partial` (boolean) is a better attribute name. 

The definition of `NetworkGraph` also needs to be improved.
